### PR TITLE
CLOUD-373 PMM2.0 prepwork

### PR DESCRIPTION
**Pull request by cap1984**

### CODE REVIEW

File: pkg/pxc/app/pmm.go [hunk 1]
```diff
@@ -7,11 +7,33 @@ import (
...
+		{
+			Name:  "CLIENT_PORT_LISTEN",
+			Value: "7777",
+		},
+		{
+			Name:  "CLIENT_PORT_MIN",
+			Value: "30100",
+		},
+		{
+			Name:  "CLIENT_PORT_MAX",
```
Comment (delgod) [inline pkg/pxc/app/pmm.go]: can we avoid CLIENT_* vars?

Comment (cap1984) [reply pkg/pxc/app/pmm.go]: I was provided with a valid env configuration in https://github.com/Percona-Lab/pmm-submodules/blob/PMM-4417-k8s-button/build/pmm-server-docker/docker-compose.yml#L8-L29

Hope to get rid of all junk in the future, however it shall be kept here for now.

File: pkg/pxc/app/pmm.go [hunk 2]
```diff
@@ -6,24 +6,56 @@ import (
...
 	pmmEnvs := []corev1.EnvVar{
 		{
 			Name:  "PMM_SERVER",
 			Value: spec.ServerHost,
 		},
 	}
 
+	clientEnvs := []corev1.EnvVar{
+		{
+			Name:  "CLIENT_NAME",
```
Comment (delgod) [inline pkg/pxc/app/pmm.go]: @nonemax please remove whole `CLIENT_NAME` var

File: deploy/cr.yaml [hunk 3]
```diff
@@ -129,10 +129,10 @@ spec:
   pmm:
     enabled: false
     image: perconalab/pmm-client:1.17.1
-    serverHost: monitoring-service
-    serverUser: pmm
+    serverHost: monitoring-service:443
+    serverUser: admin
```
Comment (delgod) [inline deploy/cr.yaml]:
```suggestion
    serverUser: pmm
```

File: pkg/pxc/app/statefulset/node.go [hunk 4]
```diff
@@ -177,12 +177,32 @@ func (c *Node) PMMContainer(spec *api.PMMSpec, secrets string) corev1.Container
 				SecretKeyRef: app.SecretKeySelector(secrets, "monitor"),
 			},
 		},
+
```
Comment (delgod) [inline pkg/pxc/app/statefulset/node.go]:
```suggestion
```

File: pkg/apis/pxc/v1/pxc_types.go [hunk 5]
```diff
@@ -163,6 +166,7 @@ type PMMSpec struct {
 	ServerHost string        `json:"serverHost,omitempty"`
 	Image      string        `json:"image,omitempty"`
 	ServerUser string        `json:"serverUser,omitempty"`
+	ServerPass string        `json:"serverPass,omitempty"`
```
Comment (delgod) [inline pkg/apis/pxc/v1/pxc_types.go]: please remove

File: pkg/pxc/app/statefulset/node.go [hunk 6]
```diff
@@ -159,8 +159,8 @@ func (c *Node) AppContainer(spec *api.PodSpec, secrets string) corev1.Container
 
 func (c *Node) SidecarContainers(spec *api.PodSpec, secrets string) []corev1.Container { return nil }
 
-func (c *Node) PMMContainer(spec *api.PMMSpec, secrets string) corev1.Container {
-	ct := app.PMMClient(spec, secrets)
+func (c *Node) PMMContainer(spec *api.PMMSpec, secrets string, availableVersion bool) corev1.Container {
```
Comment (delgod) [inline pkg/pxc/app/statefulset/node.go]:
```suggestion
func (c *Node) PMMContainer(spec *api.PMMSpec, secrets string, v120OrGreater bool) corev1.Container {
```

File: pkg/apis/pxc/v1/pxc_types.go [hunk 7]
```diff
@@ -234,7 +237,7 @@ type ServerVersion struct {
 type App interface {
 	AppContainer(spec *PodSpec, secrets string) corev1.Container
 	SidecarContainers(spec *PodSpec, secrets string) []corev1.Container
-	PMMContainer(spec *PMMSpec, secrets string) corev1.Container
+	PMMContainer(spec *PMMSpec, secrets string, availableVersion bool) corev1.Container
```
Comment (delgod) [inline pkg/apis/pxc/v1/pxc_types.go]:
```suggestion
	PMMContainer(spec *PMMSpec, secrets string, v120OrGreater bool) corev1.Container
```

**Pull request merged**